examples/decoupled/repeat_model.py

```diff
@@ -0,0 +1,206 @@
+        if len(requests) != 1:
+            raise pb_utils.TritonModelException("unsupported batch size " +
+                                                len(requests))
+
+        # Start a separate thread to send the responses for the request.
+        thread = threading.Thread(
+            target=response_thread,
+            args=(self, requests[0].get_response_sender(),
+                  pb_utils.get_input_tensor_by_name(requests[0],
+                                                    'IN').as_numpy(),
+                  pb_utils.get_input_tensor_by_name(requests[0],
+                                                    'DELAY').as_numpy()))
+        thread.daemon = True
+
+        with self.inflight_thread_count_lck:
+            self.inflight_thread_count += 1
+
+        thread.start()
+
+        # Read WAIT input for wait time, then return so that Triton can call execute again with
+        # another request.
+        wait_input = pb_utils.get_input_tensor_by_name(requests[0],
+                                                       'WAIT').as_numpy()
+        time.sleep(wait_input[0] / 1000)
+
+        return None
+
+    def response_thread(self, response_sender, in_input, delay_input):
+        # The response_sender is used to send response(s) associated with the corresponding request.
+        # Iterate over input/delay pairs. Wait for DELAY milliseconds and then create and
+        # send a response.
+
+        idx_dtype = self.idx_dtype
+        out_dtype = self.out_dtype
+
+        for idx in range(in_input.size):
+            in_value = in_input[idx]
+            delay_value = delay_input[idx]
+
+            time.sleep(delay_value / 1000)
+
+            idx_output = pb_utils.Tensor("IDX", numpy.array([idx], idx_dtype))
+            out_output = pb_utils.Tensor("OUT",
+                                         numpy.array([in_value], out_dtype))
+            response = pb_utils.InferenceResponse(
+                output_tensors=[idx_output, out_output])
+            response_sender.send(response)
+
+        # We must close the response sender to indicate to Triton that we are done sending
+        # responses for the corresponding request. We can't use the response sender after
+        # closing it.
+        response_sender.close()
+
+        with self.inflight_thread_count_lck:
+            self.inflight_thread_count -= 1
+
+    def finalize(self):
+        """`finalize` is called only once when the model is being unloaded.
+        Implementing `finalize` function is OPTIONAL. This function allows
+        the model to perform any necessary clean ups before exit.
+        Here we will wait for all response threads to complete sending
+        responses.
+        """
+        print('Finalize invoked')
+
+        inflight_threads = True
+        while inflight_threads:
+            with self.inflight_thread_count_lck:
+                inflight_threads = (self.inflight_thread_count != 0)
```

Comment:
Can you add a print statement here that prints the number of inflight threads? Similar to what we have in Triton when the responses have not been released.